### PR TITLE
Remove series flag from a bunch of CLI commands

### PR DESCRIPTION
**Pull request by jack-w-shaw**
Remove deprecated --series flag from 4.0 from `deploy`, `set-application-base`, `info`, `download`, `bootstrap`, `add-machine` and `upgrade-machine`

Instead users should use a base (--base, --bootstrap-base, or base in the positional arguement)

This is a step towards removing the concept of a series from Juju

We also still use --series in a large number of our integration tests. Fix these while I'm here

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing
- [ ] [Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing
- [ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages

## QA steps

Uni tests pass

```
./main.sh -v machine
```

```
$ juju deploy -h | grep '\-\-series'
```

```
$ juju deploy ubuntu --series jammy
ERROR option provided but not defined: --series

$ juju deploy ubuntu --base ubuntu@20.04
```

```
$ juju set-application-base ubuntu jammy
ERROR expected base string to contain os and channel separated by '@'

$ juju set-application-base ubuntu ubuntu@22.04
WARNING To ensure the correct charm binaries are installed when add-unit is next called, please first run `juju refresh` for this application and any related subordinates.
```

```
$ juju info ubuntu --series focal
ERROR option provided but not defined: --series

$ juju info ubuntu --base ubuntu@20.04
name: ubuntu
publisher: charmers
summary: A pristine Ubuntu Server
description: |
  This simply deploys the Ubuntu Cloud/Server image
store-url: https://charmhub.io/ubuntu
charm-id: DksXQKAQTZfsUmBAGanZAhpoS4dpmXel
supports: ubuntu@18.04, ubuntu@20.04, ubuntu@22.04
subordinate: false
channels: |
  latest/stable:     24  2023-05-26  (24)  2MB    amd64, arm, arm64, i386, ppc64
                     21  2022-09-15  (21)  2MB    amd64
                     19  2022-02-09  (19)  828kB  amd64, arm64, ppc64el, riscv64, s390x
                     18  2022-02-09  (18)  5MB    amd64, arm64, ppc64el, riscv64, s390x
                     17  2020-11-11  (17)  5MB    amd64, arm64, ppc64el, riscv64, s390x
  latest/candidate:  24  2023-05-19  (24)  2MB    amd64, arm, arm64, i386, ppc64
                     21  2022-09-15  (21)  2MB    amd64
                     19  2022-02-09  (19)  828kB  amd64, arm64, ppc64el, riscv64, s390x
                     18  2022-02-09  (18)  5MB    amd64, arm64, ppc64el, riscv64, s390x
                     17  2020-11-11  (17)  5MB    amd64, arm64, ppc64el, riscv64, s390x
  latest/beta:       24  2023-05-19  (24)  2MB    amd64, arm, arm64, i386, ppc64
                     21  2022-09-15  (21)  2MB    amd64
                     19  2022-02-09  (19)  828kB  amd64, arm64, ppc64el, riscv64, s390x
                     18  2022-02-09  (18)  5MB    amd64, arm64, ppc64el, riscv64, s390x
                     17  2020-11-11  (17)  5MB    amd64, arm64, ppc64el, riscv64, s390x
  latest/edge:       24  2023-05-19  (24)  2MB    amd64, arm, arm64, i386, ppc64
                     21  2022-09-15  (21)  2MB    amd64
                     19  2022-02-09  (19)  828kB  amd64, arm64, ppc64el, riscv64, s390x
                     18  2022-02-09  (18)  5MB    amd64, arm64, ppc64el, riscv64, s390x
                     17  2020-11-11  (17)  5MB    amd64, arm64, ppc64el, riscv64, s390x
```

```
$ juju download ubuntu --series focal
ERROR option provided but not defined: --series

$ juju download ubuntu --base ubuntu@20.04
Fetching charm "ubuntu" revision 24 using "stable" channel and base "amd64/ubuntu/20.04"
Install the "ubuntu" charm with:
    juju deploy ./ubuntu_r24.charm
```

```
$ juju bootstrap lxd lxd --bootstrap-series focal
ERROR option provided but not defined: --bootstrap-series

$ $ juju bootstrap lxd lxd --bootstrap-base ubuntu@20.04
Creating Juju controller "lxd" on lxd/localhost
Looking for packaged Juju agent version 4.0-beta3 for amd64
No packaged binary found, preparing local Juju agent binary
...
```

```
$ juju add-machine --series focal
ERROR option provided but not defined: --series

$ juju add-machine --base ubuntu@20.04
created machine 0

$ juju status
$ juju status
Model  Controller  Cloud/Region         Version      Timestamp
m2     lxd-src     localhost/localhost  4.0-beta3.1  12:51:42+01:00

Machine  State    Address        Inst id        Base          AZ  Message
0        pending  10.219.211.85  juju-2cbd49-0  ubuntu@20.04      Running
```

```
$ juju upgrade-machine 0 prepare jammy
ERROR expected base string to contain os and channel separated by '@'

$$ juju upgrade-machine 0 prepare ubuntu@22.04
WARNING: This command will mark machine "0" as being upgraded to "ubuntu@22.04".
This operation cannot be reverted or canceled once started.
Units running on the machine will also be upgraded. These units include:
  - ubu/0

Leadership for the following applications will be pinned and not
subject to change until the "complete" command is run:
  - ubu

Continue [y/N]? y
machine-0 validation of upgrade base from "ubuntu@20.04/stable" to "ubuntu@22.04"
machine-0 started upgrade from "ubuntu@20.04" to "ubuntu@22.04"
ubu/0 pre-series-upgrade hook running
ubu/0 pre-series-upgrade completed
machine-0 binaries and service files written

Juju is now ready for the machine base to be updated.
Perform any manual steps required along with "do-release-upgrade".
When ready, run the following to complete the upgrade base process:

juju upgrade-machine 0 complete
```

### CODE REVIEW

File: tests/suites/deploy/deploy_charms.sh [hunk 1]
```diff
@@ -170,9 +170,7 @@ run_deploy_lxd_to_machine() {
 
 	ensure "${model_name}" "${file}"
 
-	juju add-machine -n 2 --series=jammy
-
-	charm=./tests/suites/deploy/charms/lxd-profile-alt
```
Comment (nvinuesa) [inline tests/suites/deploy/deploy_charms.sh:175]: why this line gets deleted? doesn't have anything to do with the series/base changes

**Review by nvinuesa — COMMENTED** [review]
small comments, I'm running the QA steps...

File: tests/suites/deploy/deploy_charms.sh (hunk 1 above)
Comment (jack-w-shaw) [reply tests/suites/deploy/deploy_charms.sh:175]: Oops

File: cmd/juju/charmhub/download.go [hunk 2]
```diff
@@ -161,18 +154,6 @@ func (c *downloadCommand) Run(cmdContext *cmd.Context) error {
 		base corebase.Base
 		err  error
 	)
-	// Note: we validated that both series and base cannot be specified in
-	// Init(), so it's safe to assume that only one of them is set here.
-	if c.series == SeriesAll {
```
[the remark this answers is not in the document]
Comment (jack-w-shaw) [reply cmd/juju/charmhub/download.go:166]: Ta

**Review by nvinuesa — APPROVED** [review]
Thanks for the fixes, QA passes

**Pull request merged**